woocommerce/woocommerce-gateway-stripe
# Add unit tests for checkout flows

# Changes proposed in this Pull Request:

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #1858

This PR contains unit tests for the UPE checkout flow. The testing emphasis is mainly on the checkout flow with a saved payment method, but I have also added a few tests for the basic UPE checkout and UPE redirection flows. It's a little beyond the scope of this specific issue, but those tests leave a guide for future testing scope to cover more user flows. If deemed necessary, a new issue can be created for these tests (e.g. errors on intent after redirection, regular UPE checkout that saves a new payment method, tests on payment methods) if desired.

# Testing instructions

Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?

1. Pull the changes in this PR and start up your local environment with `npm run up`.
2. Run `npm run test:php`.
3. It's just that easy.

-------------------
